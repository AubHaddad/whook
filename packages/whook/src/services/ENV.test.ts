import { describe, it, beforeEach, jest, expect } from '@jest/globals';
import initENV from './ENV.js';
import type { LogService } from 'common-services';

describe('initENV', () => {
  const log = jest.fn<LogService>();
  const readFile = jest.fn<(path: string) => Promise<Buffer>>();

  beforeEach(() => {
    log.mockReset();
    readFile.mockReset();
  });

  it('should work with existing file', async () => {
    readFile.mockResolvedValueOnce(
      Buffer.from(
        `DB_PASSWORD=oudelali
DB_HOST = 'localhost'
`,
      ),
    );

    const ENV = await initENV({
      NODE_ENV: 'development',
      BASE_ENV: { KEY_BASE_ENV: 'test' },
      PROCESS_ENV: { KEY_PROCESS_ENV: 'test' },
      PWD: '/home/whoami/my-whook-project',
      log,
      readFile,
    });

    expect({
      ENV,
      logCalls: log.mock.calls.filter(([type]) => !type.endsWith('stack')),
      readFileCalls: readFile.mock.calls,
    }).toMatchSnapshot();
  });

  it('should fail with non-existing file', async () => {
    readFile.mockRejectedValueOnce(new Error('EEXISTS'));

    const ENV = await initENV({
      NODE_ENV: 'development',
      BASE_ENV: { KEY_BASE_ENV: 'test' },
      PROCESS_ENV: { KEY_PROCESS_ENV: 'test' },
      PWD: '/home/whoami/my-whook-project',
      log,
      readFile,
    });

    expect({
      ENV,
      logCalls: log.mock.calls.filter(([type]) => !type.endsWith('stack')),
      readFileCalls: readFile.mock.calls,
    }).toMatchSnapshot();
  });
});
