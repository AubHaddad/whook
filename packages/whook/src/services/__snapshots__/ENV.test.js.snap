// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initENV should fail with non-existing file 1`] = `
Object {
  "ENV": Object {
    "KEY_BASE_ENV": "test",
    "KEY_PROCESS_ENV": "test",
    "NODE_ENV": "development",
    "PWD": "/home/whoami/my-whook-project",
  },
  "logCalls": Array [
    Array [
      "info",
      "Loading the environment service.",
    ],
    Array [
      "info",
      "Using local env.",
    ],
    Array [
      "info",
      "Could not load \\".env.development\\" file.",
    ],
  ],
  "readFileCalls": Array [
    Array [
      "/home/whoami/my-whook-project/.env.development",
    ],
  ],
}
`;

exports[`initENV should work with existing file 1`] = `
Object {
  "ENV": Object {
    "DB_HOST": "localhost",
    "DB_PASSWORD": "oudelali",
    "KEY_BASE_ENV": "test",
    "KEY_PROCESS_ENV": "test",
    "NODE_ENV": "development",
    "PWD": "/home/whoami/my-whook-project",
  },
  "logCalls": Array [
    Array [
      "info",
      "Loading the environment service.",
    ],
    Array [
      "info",
      "Using local env.",
    ],
    Array [
      "info",
      "Using .env file at /home/whoami/my-whook-project/.env.development.",
    ],
  ],
  "readFileCalls": Array [
    Array [
      "/home/whoami/my-whook-project/.env.development",
    ],
  ],
}
`;
